src/ScaledPolynomialTransformFitter.cc

```diff
@@ -0,0 +1,465 @@
+        for (int p = 0, q = n; p <= n; ++p, --q, ++j) {
+            _transform._poly._xCoeffs(p, q) = solution[j];
+            _transform._poly._yCoeffs(p, q) = solution[j + packedSize];
+        }
+    }
+}
+
+void ScaledPolynomialTransformFitter::updateModel() {
+    for (auto & record : _data) {
+        record.set(
+            _keys.model,
+            _transform(record.get(_keys.input))
+        );
+    }
+}
+
+double ScaledPolynomialTransformFitter::updateIntrinsicScatter() {
+    if (!_keys.valid.isValid()) {
+        throw LSST_EXCEPT(
+            pex::exceptions::LogicError,
+            "Cannot compute intrinsic scatter on fitter initialized with fromGrid."
+        );
+    }
+    double newIntrinsicScatter = computeIntrinsicScatter();
+    float varDiff = newIntrinsicScatter*newIntrinsicScatter - _intrinsicScatter*_intrinsicScatter;
+    for (auto & record : _data) {
+        record.set(_keys.outputErr, record.get(_keys.outputErr) + varDiff*Eigen::Matrix2f::Identity());
+    }
+    _intrinsicScatter = newIntrinsicScatter;
+    return _intrinsicScatter;
+}
+
+double ScaledPolynomialTransformFitter::computeIntrinsicScatter() const {
+    // We model the variance matrix of each match as the sum of the intrinsic variance (which we're
+    // trying to fit) and the per-source measurement uncertainty.
+    // We start by computing the variance directly, which yields the sum.
+    // At the same time, we find the maximum per-source variance.  Since the per-source uncertainties
+    // are actually 2x2 matrices, we use the square of the major axis of that ellipse.
+    double directVariance = 0.0; // direct estimate of total scatter (includes measurement errors)
+    double maxMeasurementVariance = 0.0; // maximum of the per-match measurement uncertainties
+    double oldIntrinsicVariance = _intrinsicScatter*_intrinsicScatter;
+    std::size_t nGood = 0;
+    for (auto const & record : _data) {
+        if (!_keys.valid.isValid() || record.get(_keys.valid)) {
+            auto delta = record.get(_keys.output) - record.get(_keys.model);
+            directVariance += 0.5*delta.computeSquaredNorm();
+            double cxx = _keys.outputErr.getElement(record, 0, 0) - oldIntrinsicVariance;
+            double cyy = _keys.outputErr.getElement(record, 1, 1) - oldIntrinsicVariance;
+            double cxy = _keys.outputErr.getElement(record, 0, 1);
+            // square of semimajor axis of uncertainty error ellipse
+            double ca2 = 0.5*(cxx + cyy + std::sqrt(cxx*cxx + cyy*cyy + 4*cxy*cxy - 2*cxx*cyy));
+            maxMeasurementVariance = std::max(maxMeasurementVariance, ca2);
+            ++nGood;
+        }
+    }
+    directVariance /= nGood;
+
+    // Function that computes the -log likelihood of the current deltas with
+    // the variance modeled as described above.
+    auto logLikelihood = [this](double intrinsicVariance) {
+        // Uncertainties in the table right now include the old intrinsic scatter; need to
```

Comment:
Is it important to use this syntax instead of defining a named function (perhaps in the anonymous namespace)? I think the latter would be easier to read.